Prose/ProseLib/Sources/App/AppReducer.swift

```diff
@@ -24,6 +28,36 @@ public let appReducer: Reducer<
         action: /AppAction.main,
         environment: \AppEnvironment.main
     ),
+    Reducer { state, action, environment in
+        switch action {
+        case let .didLogIn(jid):
+            state.route = .main(MainScreenState(
+                sidebar: .init(credentials: .init(jid: jid))
+            ))
+
+        case let .auth(.loginResult(.success(jid))):
+            if let url = URL(string: "prose://main") {
+                return environment.openURL(url, .init())
+                    .receive(on: environment.mainQueue)
+                    .map { AppAction.didLogIn(jid: jid) }
+                    .mapError(EquatableError.init)
+                    .catch { Just(AppAction.showError($0)) }
```

Comment:
Instead of raising the error to the user, let's crash here. This would be a programmer error and the app be unusable if that path wouldn't work.

Reply:
Ok, fair enough